CMakeLists.txt

```diff
@@ -73,8 +73,18 @@ if( HAVE_HDF5 )
     list(APPEND CLOUDSC_DEFINITIONS HAVE_HDF5 )
 endif()
 
+
+### Loki
 ecbuild_find_package( NAME loki )
 
+# Add option for single-precision builds
+ecbuild_add_option( FEATURE SINGLE_PRECISION
+    DESCRIPTION "Build CLOUDSC in single precision" DEFAULT OFF
```

Comment:
I don't think this should be packed in with a general housekeeping PR, as single-precision is effectively not working for the Fortran variants. In my quick and easy experimentation I got floating point errors with Nvidia compiler on host side. Would you mind taking this to a separate PR for proper due diligence?